x-pack/plugins/observability_ai_assistant/server/service/index.ts

```diff
@@ -269,7 +269,10 @@ export class ObservabilityAIAssistantService {
         [CoreStart, { security: SecurityPluginStart; actions: ActionsPluginStart }, unknown]
       >,
     ]);
-    const user = plugins.security.authc.getCurrentUser(request);
+    const user = plugins.security.authc.getCurrentUser(request) || {
+      profile_uid: 'foobar',
+      username: 'kibana_system',
+    };
```

Comment:
As discussed over slack, we should make this a system connector to have auth access. Right now the workflow is mocking a kibana request and adding nasty fallbacks to workaround lack of proper auth